framework/src/auxkernels/FunctorVectorMatPropElementalAux.C

```diff
@@ -0,0 +1,43 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#include "FunctorVectorMatPropElementalAux.h"
+#include "metaphysicl/raw_type.h"
+
+registerMooseObject("MooseApp", FunctorVectorMatPropElementalAux);
+registerMooseObject("MooseApp", FunctorADVectorMatPropElementalAux);
+
+template <bool is_ad>
+InputParameters
+FunctorVectorMatPropElementalAuxTempl<is_ad>::validParams()
+{
+  InputParameters params = AuxKernel::validParams();
+  params.addClassDescription(
+      "Evaluates a functor vector material property on the current element."
+      "For finite volume, this evaluates the material property at the centroid.");
+  params.addRequiredParam<MaterialPropertyName>("mat_prop", "The functor material property");
```

Comment:
I started this myself, but we can move away from implying what kind of functor this is, e.g. this aux kernel (well the AD version of it) would work equally well with a functor that is in fact a MOOSE vector variable. Then this parameter type would be our generic `MooseFunctorName`, and the member name could be more general. In my interpolator PR I'll be adding more types of functors and it's hard to anticipate what more we will add in the future. With that in mind, we don't need to restrict the notion of what our inputs are. (Also in the interpolator PR I actually add the ability to change the backing functor through the course of the simulation and at the same time allow consumers to continue to hold a valid reference using the letter-and-envelope idiom)

Reply:
yes but as you said we ve kept on the traditional naming (function / matprop / auxvars) so far.

If I drop this here I ll refact all our objects then.

Reply:
I'm admittedly based on a commit that's a little old but I don't see that many objects that have specific meaning
```
(base) lindad@inl610212:~/projects/moose(insfv-interpolator)$ git grep -n "getFunctor.*mat"
framework/src/auxkernels/FunctorMatPropElementalAux.C:31:  : AuxKernel(parameters), _mat_prop(getFunctor<GenericReal<is_ad>>("mat_prop"))
modules/navier_stokes/src/postprocessors/InternalVolumetricFlowRate.C:52:    _advected_material_property(getFunctor<ADReal>("advected_mat_prop"))
modules/navier_stokes/src/postprocessors/VolumetricFlowRate.C:52:    _advected_material_property(getFunctor<ADReal>("advected_mat_prop"))
test/src/bcs/MatPropBC.C:24:  : ADIntegratedBC(parameters), _coef(getFunctor<ADReal>("mat_prop"))
test/src/fvkernels/FVFunctorElementalKernel.C:25:  : FVElementalKernel(params), _functor_prop(getFunctor<ADReal>("mat_prop_name"))
```
Of the production objects, I would contend that only `FunctorMatPropElementalAux` is poorly named

Reply:
It's a shame about the parameter names for `(Internal)VolumetricFlowRate` ... if they were just `advected_prop` then I think there's no connotation that this is a material property. Even now, the parameter type for those is `MooseFunctorName`

Reply:
there s also the VarMatFunctorAux or something.

Reply:
I agree it s not a lot. Which is why I m offering to do it ^^

Reply:
I couldn't tell whether you were being sassy or not 😆 Hard to tell over a comp. If you're offering, then I think that would be awesome

Reply:
ok I m gonna keep the two volumetric flow rate ones for when I work on SAM coupling. I think I ll see more clearly what the needs are then. I suspect we will be able to fuse the advected_variable and advected_matprop into an advected_functor parameter, but i d like to be sure